Utilities/XrdAdaptor/src/XrdFile.cc

```diff
@@ -248,9 +250,54 @@ XrdFile::read (void *into, IOSize n, IOOffset pos)
     addConnection(ex);
     throw ex;
   }
+  if (n == 0) {
+    return 0;
+  }
+
+  // In some cases, the IO layers above us (particularly, if lazy-download is
+  // enabled) will emit very large reads.  We break this up into multiple
+  // reads in order to avoid hitting timeouts.
+  std::vector<IOPosBuffer> requests;
```

Comment:
You can also do a reserve call here by using the integer division if n and XRD_CL_MAX_READ_SIZE